File: sprint1/By languages/Java/lesson6code2.java
int[] cumulativeSums = new int[sequence.length() + 1];
int cgCount = 0;

for (int i = 0; i < sequence.length(); i++) {
    if (sequence.charAt(i) == 'C' || sequence.charAt(i) == 'G') {
        cgCount++;
    }
    cumulativeSums[i + 1] = cgCount;
}

System.out.println(cumulativeSums[right] - cumulativeSums[left]);

File: sprint1/By languages/kotlin/lesson6code2.kt
Copy code
var cumulativeSums = mutableListOf(0)
var cgCount = 0
for (position in 0 until sequence.length) {
    if (sequence[position] == 'C' || sequence[position] == 'G') {
        cgCount++
    }
    cumulativeSums.add(cgCount)
}

println(cumulativeSums[right] - cumulativeSums[left])
